xray/core/dataarray.py

```diff
@@ -1052,6 +1052,39 @@ def func(self, other):
             return self
         return func
 
+    def diff(self, dim, n=1):
```

Comment:
can we add `coord_new` (or `label`) to the function signature here, too?

This function could also use at least one test to verify that it works.